tests/modeltests/tst_models.cpp

```diff
@@ -25,6 +25,12 @@
   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <cxxabi.h>
```

Comment:
sorry, you misunderstood my intentions. your test is good in coverage but its implementation is overly complicated. Instead, just add a `test_prettifySymbol_data()` and `test_prettifySymbol()`. In the `_data`, add two `QString` columns, one with the "raw" symbol and one with the extected pretty symbol. In the other function, just fetch the two strings, call  prettify on the input and then compare the result to the expected value.
This will also make it trivial to add tests for more complex symbols with high levels of template nesting. Please do test that too, something like:

```
std::__1::__function::__func<std::__1::__bind<bool (foobar::map::api_v2::DeltaAccessImpl::*)(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::weak_ptr<stream::Downloader> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), foobar::map::api_v2::DeltaAccessImpl*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::weak_ptr<stream::Downloader> const&, char const (&) [1], char const (&) [1], std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>, std::__1::allocator<std::__1::__bind<bool (foobar::map::api_v2::DeltaAccessImpl::*)(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::weak_ptr<stream::Downloader> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&), foobar::map::api_v2::DeltaAccessImpl*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::weak_ptr<stream::Downloader> const&, char const (&) [1], char const (&) [1], std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&> >, bool ()
```

Yes, that is a real-world example for a symbol. You can also have a look at the symbols in `nm -aDC tests/test-clients/cpp-parallel/cpp-parallel` for more inspiration of complex symbols. Actually, it would be really cool if you could ensure that the symbols encountered on the hot path in that example are prettified properly:

```
    99.92%     0.00%  cpp-parallel  libc-2.29.so         [.] __GI___clone (inlined)
            |
            ---__GI___clone (inlined)
               |          
                --99.92%--start_thread
                          |          
                           --99.92%--execute_native_thread_routine
                                     std::thread::_State_impl<std::thread::_Invoker<std::tuple<std::__future_base::_Async_state_impl<std::thread::_Invoker<std::tuple<double (*)()> >, doubl>
                                     std::thread::_Invoker<std::tuple<std::__future_base::_Async_state_impl<std::thread::_Invoker<std::tuple<double (*)()> >, double>::_Async_state_impl(std>
                                     std::thread::_Invoker<std::tuple<std::__future_base::_Async_state_impl<std::thread::_Invoker<std::tuple<double (*)()> >, double>::_Async_state_impl(std>
                                     std::__invoke<std::__future_base::_Async_state_impl<std::thread::_Invoker<std::tuple<double (*)()> >, double>::_Async_state_impl(std::thread::_Invoker<>
                                     std::__invoke_impl<void, std::__future_base::_Async_state_impl<std::thread::_Invoker<std::tuple<double (*)()> >, double>::_Async_state_impl(std::thread>
                                     std::__future_base::_Async_state_impl<std::thread::_Invoker<std::tuple<double (*)()> >, double>::_Async_state_impl(std::thread::_Invoker<std::tuple<dou>
                                     std::__future_base::_State_baseV2::_M_set_result (inlined)
                                     std::call_once<void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_>
                                     __gthread_once (inlined)
                                     __pthread_once_slow
                                     |          
                                      --99.92%--std::__future_base::_State_baseV2::_M_do_set
                                                std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>::operator() (inlined)
                                                std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base:>
                                                std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<double>, std::__future_base::_Result_base::_Deleter>, std::thre>
                                                std::thread::_Invoker<std::tuple<double (*)()> >::operator() (inlined)
                                     std::__future_base::_State_baseV2::_M_set_result (inlined)
                                     std::call_once<void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_>
                                     __gthread_once (inlined)
                                     __pthread_once_slow
                                     |          
                                      --99.92%--std::__future_base::_State_baseV2::_M_do_set
                                                std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>::operator() (inlined)
                                                std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> (), std::__future_base:>
                                                std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<double>, std::__future_base::_Result_base::_Deleter>, std::thre>
                                                std::thread::_Invoker<std::tuple<double (*)()> >::operator() (inlined)
                                                std::thread::_Invoker<std::tuple<double (*)()> >::_M_invoke<0ul> (inlined)
                                                std::__invoke<double (*)()> (inlined)
                                                std::__invoke_impl<double, double (*)()> (inlined)
                                                worker
                                                |          
                                                |--96.22%--std::uniform_real_distribution<double>::operator()<std::linear_congruential_engine<unsigned long, 16807ul, 0ul, 2147483647ul> > (>
                                                |          std::uniform_real_distribution<double>::operator()<std::linear_congruential_engine<unsigned long, 16807ul, 0ul, 2147483647ul> > (>
                                                |          |          
                                                |           --88.87%--std::__detail::_Adaptor<std::linear_congruential_engine<unsigned long, 16807ul, 0ul, 2147483647ul>, double>::operator(>
                                                |                     std::generate_canonical<double, 53ul, std::linear_congruential_engine<unsigned long, 16807ul, 0ul, 2147483647ul> > (in>
                                                |                     |          
                                                |                      --53.43%--std::linear_congruential_engine<unsigned long, 16807ul, 0ul, 2147483647ul>::operator() (inlined)
                                                |                                std::__detail::__mod<unsigned long, 2147483647ul, 16807ul, 0ul> (inlined)
                                                |                                std::__detail::_Mod<unsigned long, 2147483647ul, 16807ul, 0ul, true, true>::__calc (inlined)
                                                |          
                                                 --2.39%--std::norm<double> (inlined)
                                                           std::_Norm_helper<true>::_S_do_it<double> (inlined)
```

The output above was generated with `perf --stdio`, using binutils, so it may be slightly different from what we get from elfutils in hotspot. But I guess you get the idea.

Reply:
What c++ code generated the "real world" symbol `std::__1::__function::__func<std::__1::__bind<bool ...`? I'm interested in replicating it.

Reply:
that was a symbol reported to me by a colleague from a customer code base we aren't allowed to share.

anyhow, for the purpose of this test, I don't see the need to replicate it using actual C++ code - simplifying operates on strings anyways, so use it as an input and see if it's properly simplified

Reply:
I managed to figure it out. It has been added to the unit test.